etl/steps/data/meadow/cedlas/2024-03-08/sedlac.py

```diff
@@ -0,0 +1,351 @@
+
+    #
+    # Save outputs.
+    #
+    # Create a new meadow dataset with the same metadata as the snapshot.
+    ds_meadow = create_dataset(
+        dest_dir,
+        tables=income_tables + inequality_deciles_indices_tables + inequality_gini_tables + poverty_tables,
+        check_variables_metadata=True,
+        default_metadata=snap_poverty.metadata,
+    )
+
+    # Save changes in the
+    # new meadow dataset.
+    ds_meadow.save()
+
+
+def load_tables_from_snapshot(snap: Snapshot, sheets: Dict) -> List[Table]:
+    """Load all the sheets from the snapshot."""
+    tables = []
+    for sheet in sheets:
+        tb = snap.read(sheet_name=sheet, header=sheets[sheet]["header"])
+        tb.metadata.short_name = sheets[sheet]["short_name"]
+        tables.append(tb)
+    return tables
+
+
+def format_income_tables(tb: List[Table], aggregation: List[str]) -> List[Table]:
+    """Format the income table."""
+
+    tables = []
+    for t in tb:
+        # Fill columns with data for spells
+        t.loc[0] = t.loc[0].fillna(method="ffill")
+
+        # Flatten column names
+        t.columns = ["_".join(map(str, x)) if isinstance(x, tuple) else str(x) for x in t.columns.values]
+
+        # Rename columns as the current name + the first row
+        t.columns = t.columns + "_" + t.loc[0].astype(str)
+
+        # Remove the first row
+        t = t.drop(0).reset_index(drop=True)
+
+        # Remove spaces between column names
+        t.columns = t.columns.str.replace(" ", "")
+
+        # Rename first column to "indicator"
+        t = t.rename(columns={"Country_Year_nan": "indicator"})
+
+        # Create column aggregation, by selecting indicator where values are [NATIONAL, URBAN, RURAL]. Fill empty values with the previous value
+        t["aggregation_level"] = t.loc[t["indicator"].isin(aggregation), "indicator"]
+        t["aggregation_level"] = t["aggregation_level"].fillna(method="ffill")
+
+        # Drop rows with indicator = null
+        t = t.dropna(subset=["indicator"]).reset_index(drop=True)
+
+        # Drop rows with indicator = [NATIONAL, URBAN, RURAL]
+        t = t[~t["indicator"].isin(aggregation)].reset_index(drop=True)
+
+        # If values in indicator are between 1 and 10, add "decile" to the indicator
+        t.loc[t["indicator"].astype(str).str.isnumeric(), "indicator"] = "decile" + t["indicator"].astype(str) + "_thr"
+
+        # Remove last row if it is SOURCE_TEXT
+        assert t.iloc[-1, 0] == SOURCE_TEXT, f"Last row is not {SOURCE_TEXT}. Check the file."
+
+        t = t.drop(t.index[-1]).reset_index(drop=True)
+
+        # Make table wider, using aggregation_level as a column
+        t = t.pivot(index="indicator", columns="aggregation_level").reset_index()
+
+        # Flatten column names
+        t.columns = ["_".join(map(str, x)) if isinstance(x, tuple) else str(x) for x in t.columns.values]
+
+        # Transpose the table
+        t_transposed = t.copy().T.reset_index()
```

Comment:
Hi @pabloarosado! What would be a way to keep metadata for the columns in this table I am transposing? I tried reassigning the metadata from the previous table but that doesn't seem to work.

Reply:
Pablo, no worries about this, I won't process this data after all.